qucs/qucs/qucs.cpp

```diff
@@ -834,44 +860,37 @@ void QucsApp::slotSelectComponent(QListWidgetItem *item)
   char *CompFile_cptr;
 
   if (!CompSearch->text().isEmpty()) {
-    //comp search is not empty, search all category to get a component in same name
-    QStringList cats = Category::getCategories();
-    int i = 0;
-    //search normal component
-    foreach (QString it, cats) {
-      Comps = Category::getModules (it);
-      foreach(Module *mod, Comps) {
-        if (mod->info) {
-          (*mod->info)(CompName, CompFile_cptr, false);
-          if (CompName == name) {
-	    // change currently selected category, so the user will 
-	    //   learn where the component comes from
-            CompChoose->setCurrentIndex(i+1); // +1 due to the added "Search Results" item
-	    ccCurIdx = i; // remember the category to select when exiting search
-	    //!! comment out the above two lines if you would like that the search
-	    //!!   returns back to the last selected category instead
-            view->selElem = (*mod->info) (CompName, CompFile_cptr, true);
-	    // TODO: component found, exit the loops now...
-          }
-        }
+    // in "search mode"
+    QVariant v = CompComps->item(i)->data(Qt::UserRole);
+    iconCompInfoStruct iconCompInfo = v.value<iconCompInfoStruct>();
+    qDebug() << "variant :" << v << iconCompInfo.catIdx << iconCompInfo.compIdx;
+
+    Category* cat = Category::Categories.at(iconCompInfo.catIdx);
+    Module *mod = cat->Content.at(iconCompInfo.compIdx);
+    qDebug() << "mod->info" << mod->info;
+    qDebug() << "mod->infoVA" << mod->infoVA;
+    Infos = mod->info;
```

Comment:
weird, GitHub still says you *commented on an outdated diff* but this time I did not rebase :wink: . I also used a different branch from the one used previously to avoid these issues... go figure!

I think you were referring to [this](https://github.com/Qucs/qucs/pull/547#discussion_r71971661). If this means modifying [module.cpp](https://github.com/Qucs/qucs/blob/release-0.0.19/qucs-core/src/module.cpp) I'm a bit reluctant to do that as I do not really understand its inner workings. That part I practically reused from the existing code.